lib/compat/wordpress-5.9/class-wp-theme-json-5-9.php

```diff
@@ -1470,7 +1495,7 @@ public function merge( $incoming ) {
 
 			// Replace the presets.
 			foreach ( static::PRESETS_METADATA as $preset ) {
-				$override_preset = static::should_override_preset( $this->theme_json, $node['path'], $preset['override'] );
+				$override_preset = ! static::get_metadata_boolean( $this->theme_json['settings'], $preset['prevent_override'], true );
```

Comment:
here's an example of the confusion: `$override_preset` reads to me like a directive, such as "you should override the preset" while `prevent_override` sounds more like an allowance, "you may or may not override the preset."

two changes that may not be ideal but which would remove the confusion from this line would be to:

```php
// keep the name so it means the same thing when reading
$prevent_override = static::get_metadata_boolean( … );

…

		( 'theme' === $origin && ! $prevent_override )
```

```php
// mirror the permissive language of the setting from the JSON file
// but in the positive sense
$allow_override = ! static::get_metatdata_boolean( … );

…

		( 'theme' === $origin && $allow_override )
```